### PR TITLE
Allow Credentials in POST

### PR DESCRIPTION
**Pull request by danygielow**
client_id and client_secret can also be given via POST data. So added client_id and client_secret to Post namedtuple and trying to retrieve them via POST first. Auth Header if they were not found.

https://www.oauth.com/oauth2-servers/access-tokens/client-credentials/

### CODE REVIEW

**Comment by aliev** [comment]
@danygielow thank you for another great PR! I will take a look this evening, we have small issues in typing: https://github.com/aliev/aioauth/pull/24/checks?check_run_id=2001875113

**Comment by synchronizing** [comment]
Seems good with me.

Note on [RFC6749 Section 3.2, Token Endpoint](https://tools.ietf.org/html/rfc6749#section-3.2): 
> The client MUST use the HTTP "POST" method when making access token requests.

File: src/aioauth/grant_type.py [hunk 1]
```diff
@@ -68,6 +68,17 @@ async def validate_request(self, request: Request) -> Client:
 
         return client
 
+    def get_client_credentials(
+        self, request: Request
+    ) -> Tuple[Optional[str], Optional[str]]:
```
Comment (synchronizing) [inline src/aioauth/grant_type.py:73]: `-> Tuple[str, str]:` should fix the current issue with the auto-checker.

**Comment by synchronizing** [comment]
All seems resolved. @aliev Everything looks good to me. 👍🏼

File: src/aioauth/grant_type.py (hunk 1 above)
Comment (danygielow) [reply src/aioauth/grant_type.py]: Hi, I changed the return type as you suggested. Thank you.
The project should have a type checker in pre-commit-config to catch these problems before opening a PR. Maybe add a mypy.ini and a mypy section to pre-commit-config or a pyright equivalent.

**Review by aliev — APPROVED** [review]

**Comment by aliev** [comment]
for me looks great as well :) @danygielow great job thank you! approved

**Pull request merged**